unstructured_ingest/v2/processes/connectors/vectara.py

```diff
@@ -0,0 +1,323 @@
+import json
+import uuid
+from dataclasses import dataclass, field
+from datetime import date, datetime
+from pathlib import Path
+from typing import Any, Dict, List, Mapping, Optional
+
+from dateutil import parser
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import DestinationConnectionError
+from unstructured_ingest.utils.data_prep import flatten_dict
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    FileData,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import DestinationRegistryEntry
+
+BASE_URL = "https://api.vectara.io/v1"
+
+CONNECTOR_TYPE = "vectara"
+
+
+class VectaraAccessConfig(AccessConfig):
+    oauth_client_id: Optional[str] = Field(default=None, sensitive=True, description="Client ID")
+    oauth_secret: Optional[str] = Field(default=None, sensitive=True, description="Client Secret")
+
+
+class VectaraConnectionConfig(ConnectionConfig):
+    access_config: Secret[VectaraAccessConfig]
+    customer_id: str
+    corpus_name: Optional[str] = None
+    corpus_id: Optional[str] = None
+    token_url: str = "https://vectara-prod-{}.auth.us-west-2.amazoncognito.com/oauth2/token"
+
+
+class VectaraUploadStagerConfig(UploadStagerConfig):
+    pass
+
+
+@dataclass
+class VectaraUploadStager(UploadStager):
+    upload_stager_config: VectaraUploadStagerConfig = field(
+        default_factory=lambda: VectaraUploadStagerConfig()
+    )
+
+    @staticmethod
+    def parse_date_string(date_string: str) -> date:
+        try:
+            timestamp = float(date_string)
+            return datetime.fromtimestamp(timestamp)
+        except Exception as e:
+            logger.debug(f"date {date_string} string not a timestamp: {e}")
+        return parser.parse(date_string)
+
+    @staticmethod
+    def conform_dict(data: dict) -> dict:
+        """
+        Prepares dictionary in the format that Vectara requires
+
+        Select which meta-data fields to include and optionally map them to a new new.
+        remove the "metadata-" prefix from the keys
+        """
+        metadata_map = {
+            "page_number": "page_number",
+            "data_source-url": "url",
+            "filename": "filename",
+            "filetype": "filetype",
+            "last_modified": "last_modified",
+        }
+        md = flatten_dict(data, separator="-", flatten_lists=True)
+        md = {k.replace("metadata-", ""): v for k, v in md.items()}
+        md = {metadata_map[k]: v for k, v in md.items() if k in metadata_map}
+        return md
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        docs_list: Dict[Dict[str, Any]] = []
```

Comment:
This is annotated as a dict yet it's set to a list. I'm assuming this should be:
```suggestion
        docs_list: list[dict[str, Any]] = []
```

It's also defined here and again on line 96.